ftw/monitor/autowarmup.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import os
+import requests
+import threading
+
+
+log = logging.getLogger('ftw.monitor')
+
+
+def autowarmup(port):
+    """Send a warmup request to a booting instance in a non-blocking way.
+    """
+    log.info("Autowarmup starting...")
+
+    # Fire off a daemon thread that triggers the warmup in order to not block
+    # instance startup. Doing a blocking request here would deadlock.
+    warmup_thread = threading.Thread(
+        name='warmup-%s' % port,
+        target=trigger_warmup,
+        args=(port,))
+    warmup_thread.setDaemon(True)
+
+    # Start the thread. No need to ever join it - it will either terminate on
+    # its own once the warmup response is received, or, since it's a daemon
+    # thread, will be killed when the main thread exits.
+    warmup_thread.start()
+
+
+def trigger_warmup(port):
+    """Dispatch a warmup request to the @@warmup view on the Zope app root.
+    """
+    url = 'http://localhost:%s/@@warmup' % port
+
+    # Time until ZServer starts accepting connections on the socket.
+    # With the fast-listen option this is usually very quick, but we
+    # still allow for a reasonable grace period.
+    connect_timeout = 20
```

Comment:
As discussed: I'd rather not, since this shouldn't need to be customized by now. And if it doesn, I'd rather make it _properly_ customizable, instead of encouraging monkey-patching.